### ISSUE TITLE
10.3 - Changelog test Results

### ISSUE THREAD

**Issue by davitol** [owncloud/core#36142]
### Added
- [x] Support for php 7.3 #34559 #35775 #35752 @phil-davis PR #36148 demonstrates acceptance tests with PHP  7.3
- [x] Support for redirecting private links to ownCloud phoenix frontend #35819 @individual-it 
- [ ]  encryption:fixencryptedversion command to address issues related to encrypted versions https://github.com/owncloud/encryption/pull/115 :x: functionality is OK, but output is not good https://github.com/owncloud/encryption/issues/146 @individual-it 
- [x] Tech preview DAV endpoint for public shares #35932 #36057 #36021 #36059 #36066 #36080 #36061 #36119 #36049 #36068 @phil-davis tested by CI, endpoints are disabled by default by #36124 
- [x] Tech preview DAV endpoint for trashbin #35716 #35879 #36053 #36073 @phil-davis tested by CI, endpoints are disabled by default by #36124
- [x] OCS Roles API and ability to set permissions via share attributes - #36024 #36086 @individual-it (not tested every possible security issue, need automated tests)
- [x] OCS API for public link share email notification - #36063 :x: #36156 #36157 Fixed in RC1
- [x] JS API v2 for share attributes - #35836 @phil-davis https://github.com/owncloud/richdocuments/pull/276 demonstrates this passing against 10.3.0alpha2 
- [x] Url /cron in addition to /cron.php to execute cronjobs via webcron #34932 @individual-it 
- [x] system:cron occ command for executing background tasks via system cron #34932 @individual-it 
- [x] previews_path config option to configure thumbnail storage path #35131 @individual-it 
- [x] Show activity when share receiver unshares a received share #35193 @HariBhandari07 
- [x] Document phoenix.baseUrl in config.sample.php - #36007 @individual-it typos fixed in #36152

**Comment by davitol** [comment]
### Changed

- [ ] Allow two-factor providers to display custom challenge message #34848
- [ ] Handling of unauthenticated ajax requests to prevent browser issues #36003
- [ ] Improved share permission handling #35884
- [ ]  Refined user administration setting button #35877
- [ ] Improved mobile device experience #35919 #35813 #35347
- [ ] Reference the new iOS app in the list of available applications #35918
- [ ] Improved sharing autocomplete dropdown layout #35397
- [ ]  Improved theming capabilities by allowing html for Name and LogoClaim #35273
- [ ] Improved private link UX for large resolutions #34998
- [ ] Improved wording for several user/administrator encryption related interactions #21 #117
- [ ] Handling of composer autoloader for apps/files_external #35755
- [ ] Renamed share icon to be adblock friendly #35199
- [ ] Bump sabre/xml 2.1.2 from to 2.1.3 #36036
- [ ] Bump sabre/dav from 3.2 to 4.0 #34559
- [ ] Bump symfony/polyfill components from v1.11.0 to v1.12.0 #35989
- [ ] Bump @bower_components/handlebars from v4.1.1 to v4.1.2 #35025
- [ ] Bump @bower_components/jsTimezoneDetect from 1.0.5 to v1.0.6 #33776
- [ ] Bump doctrine/lexer from v1.0.1 to 1.0.2 #35625
- [ ] Bump egulias/email-validator from 2.1.7 to 2.1.11 #35341 #35625 #35934 #36026 #36026
- [ ] Bump icewind/smb from 3.1.1 to 3.1.2 #36017
- [ ] Bump icewind/smb from 3.1.1 to 3.1.2 in /apps/files_external/3rdparty - #36017
- [ ] Bump league/flysystem from 1.0.51 to 1.0.55 #35275 #35644 #36099
- [ ] Bump nikic/php-parser from 4.2.1 to 4.2.3 #35337 #36015 #36015
- [ ] Bump phan to 1.3.5 and enable on PHP 7.2 7.3 - #35818
- [ ] Bump phpseclib/phpseclib from 2.0.15 to 2.0.21 #35336 #35565 #35643 #35827
- [ ] Bump league/flysystem from 1.0.51 to 1.0.53 #35275 #35644
- [ ] Bump symfony from v3.4.26 to v3.4.30 #35146 #35348 #35625 #35934
- [ ] Bump sabre/dav from 3.2 to 4.0.1 #34559 #36094
- [ ] Bump sabre/xml 2.1.2 from to 2.1.3 #36036 #36036
- [ ] Bump swiftmailer/swiftmailer from 6.2.0 to 6.2.1 #35075
- [ ] Bump symfony from v3.4.26 to v3.4.31 #35146 #35348 #35625 #35934 #36098 #36097 #35989 - Bump symfony/process from 3.4.30 to 3.4.31 - #36095 #36096 #36093
- [ ] Bump theseer/tokenizer from 1.1.2 to 1.1.3 #35625
- [ ] Updating webmozart/assert (1.4.0 => 1.5.0) - #36103
- [ ] Updating zendframework/zend-filter (2.9.1 => 2.9.2) - #36102
- [ ] Updating zendframework/zend-inputfilter (2.10.0 => 2.10.1) - #36112
- [ ] Update the minimum required Node engine version to 8.15.0 - #36033

**Comment by davitol** [comment]
### Removed

- [ ] Deprecated update script from files app #35781
- [ ]  Dropped APC and XCache support #35782
- [ ]  Old table repair step will drop deprecated contacts_cards_properties table #35721
- [ ] Removed support for swift as primary / external storage #35951
- [ ] Moved S3 external integration into separate app (files_external_s3) #34986
- [ ]  Moved ownCloud default encryption app into separate repository #35949

**Comment by davitol** [comment]
### Fixed

- [ ]  Use bit operators when checking share file permission - #36111
- [ ]  Only share owner should be able to update or delete share - #36120
- [ ]  Fix various issues with session handling in relation to redis - #35888
- [ ]  Fix issue where IE did not redirect to login page when user is not logged in - #36079
- [ ] Check that all user mount points has unique names - #36029
- [ ] Fix loading of app.php when using a separate apps folder - #36054
- [ ] Respect default app config within the TwoFactorChallengeController - #36031
- [ ] Don't send WWW-Authenticate headers with schema Basic for ajax requests - #36003
- [ ]  Fix issue when share folder and shares go missing when storage becomes unavailable - #35998
- [ ] Handling of OCM sharing when receiving server did not include a protocol (i.e. https) #35711
- [ ]  Performance improvements when loading groups of users #35822
- [ ] Relative path handling for files:checksums:verify occ command #35694
- [ ] Failed rename operation leading to unavailable external storage #35598
- [ ] Comment creation event missing ID field #35799
- [ ] Improved handling of share expire input fields to avoid user error #35779
- [ ] Refactored user creation in order to allow same flow for occ and api #35972
- [ ] Maintain dav properties when files are moved to trashbin #35954
- [ ]  Usage of domain when authenticate with SMB/WND shares #35892
- [ ] Triggering dav events on the public webdav endpoint #35820
- [ ] Prevent deletion of configured share_folder #35998
- [ ] Issues with improper displayed languages #35973
- [ ] Respect user.min_search_length with federated sharing #35977
- [ ] Avoid password manager autocomplete on user administration #35931
- [ ] Changing config settings produced duplicate emitted events #35875
- [ ] Properly return StorageNotAvailable on network failures with external storages #35707
- [ ] Improved error message when trying to share with a non-existing federated user #35542
- [ ] Allow selection of UI errors during web-installation #35681
- [ ] Added missing events for webdav copy operations on new endpoint #35604
- [ ] Double-appearing address book entries when shared with groups #35603
- [ ] Issues with federation when proxy requires credentials #35868
- [ ] Respect share_folder with federated shares #35396
- [ ] Issues with sqlite to mysql migration with db:convert-type #35390
- [ ] Upload issues with mismatching checksums #35294
- [ ] Improved memory handling for trashbin expiry background job #35708
- [ ] Proper handling of objecstorage S3 issues on object upload for files_primary_s3 core#35389 files_primary_s3#212
- [ ] Respect default application configuration when using TwoFactor Authentication #36031
- [ ] Improved mobile view for file drop links #34803
- [ ] Ignore case of userid in occ files:scan command #35324
- [ ] Properly handle errors from remote server when declining a non-existing federated share #35321
- [ ] UI issues on setup page when mobile devices where used #35347
- [ ]  Direct access to sharing tab for long file listings #35306
- [ ] Improved OCM compliance on providerId and remoteId fields #35122
- [ ] Issue with adding multiple Google Drive external storages #34987
- [ ] Issues with recreating masterkeys when HSM is used #128

**Comment by micbar** [comment]
# Fixed in RC1
## Changelog testing on RC1

- [ ] Bump sabre uri - [#36189](https://github.com/owncloud/core/issues/36189)
- [ ]  shareowner can edit re-share permissions without any restriction - [#36193](https://github.com/owncloud/core/issues/36193)
- [ ] Improve logging when a remote host went down suddenly - [#36180](https://github.com/owncloud/core/issues/36180)
- [ ] Updating sabre/http (5.0.0 => 5.0.2) - [#36192](https://github.com/owncloud/core/issues/36192)
- [ ] Add getReshareAttributes method to shareitemmodel with fix for parsing - [#36186](https://github.com/owncloud/core/issues/36186)
- [ ] Remove hardcoded protocol. - [#36127](https://github.com/owncloud/core/issues/36127)
- [ ] obey to config in share mail notifications APIs - [#36161](https://github.com/owncloud/core/issues/36161)
- [ ] Fix invalid foreach in SharingBlacklist.php - [#36038](https://github.com/owncloud/core/issues/36038)
- [ ] Improve share_folder checks in rmdir - [#36170](https://github.com/owncloud/core/issues/36170)
- [ ] Add new migrations to dav app - [#36084](https://github.com/owncloud/core/issues/36084)
- [ ] Disable trashbin and public webdav API by default - [#36124](https://github.com/owncloud/core/issues/36124)
- [ ] Dont check the CSRF token on public link email API - [#36158](https://github.com/owncloud/core/issues/36158)
- [ ] fixed typos in 'phoenix.baseUrl' documentation - [#36152](https://github.com/owncloud/core/issues/36152)
- [ ] we can't invalidate the token if there isn't a user session active - [#36153](https://github.com/owncloud/core/issues/36153)

**Comment by micbar** [comment]
@HanaGemela @jesmrec @davitol 
I posted the fixes for 10.3.0RC1
We can do the changelog testing of RC1 as soon as we are through with alpha2

**Comment by micbar** [comment]
## Client Related
- [ ]  shareowner can edit re-share permissions without any restriction - [#36193](https://github.com/owncloud/core/issues/36193)
- [ ] Fix invalid foreach in SharingBlacklist.php - [#36038](https://github.com/owncloud/core/issues/36038)
- [ ] Improve share_folder checks in rmdir - [#36170](https://github.com/owncloud/core/issues/36170)

**Comment by jesmrec** [comment]
## Mobile clients tests

_____

**shareowner can edit re-share permissions without any restriction - #36193**

Tested with 10.2.1, problem is not reproducible. With 10.3RC1, owner of an item can edit permissions of the recipient groups, even if the group is recipient of a re-share.

Steps defined in https://github.com/owncloud/core/issues/36107#issue-485856327 were performed.

_____

**Fix invalid foreach in SharingBlacklist.php**

This values were tested as blacklisted groups:

```
0
1
1.23
1.23e-4
1.098.345
1.4567e-1.23
```

neither of them were listed as backlisted, and correctly listed when they weren't blacklisted.

_____

**Improve share_folder checks in rmdir - #36170**

By adding `'share_folder' => '/Shared Folder'` to `config.php`

and following steps: https://github.com/owncloud/core/pull/36170#issuecomment-536947486

both Android and iOS do not allow to remove the `Shared Folder` folder in the clients side. Shared folder stays in both sides (client and server).

_____

**Android:** 
   App version: 2.13 (stable - [87ce8f0](https://github.com/owncloud/android/tree/oc-android-2.13))
   Device: Nexus 6P, Android 7.1

**iOS:** 
   App version: 1.1 (stable - [1e1f5da](https://github.com/owncloud/ios-app/tree/version_1.1.0))
   Device: iPhoneX, iOS13.1

**Server:** 10.3.0 RC1 (testing) 


Please, let me know if you miss something.

**Comment by HanaGemela** [comment]
I've tested the below items on the desktop sync client. All passed. I've added a comment with more details into each issue.
-  shareowner can edit re-share permissions without any restriction - #36193
-  Fix invalid foreach in SharingBlacklist.php - #36038
-  Improve share_folder checks in rmdir - #36170 

Newly found issues: 
share_folder can be shared https://github.com/owncloud/core/issues/36241
Deleted folder is not restored when deletion is not allowed https://github.com/owncloud/client/issues/7496
